### PR TITLE
S111 more api tests

### PR DESCRIPTION
**Pull request by eschultink**
More testing; no actual logic changes, so not urgent to actually get this merged.  It just validates that strengthened rules aren't blocking anything that we need in practice (eg, retrieval of subsequent pages).

### Features
  - more test coverage that rules allow query params, which is needed to support paging
  - more test coverage that rules allow/block expected url variants 

### Change implications

 - dependencies added/changed? **no**

### CODE REVIEW

**Review by aperez-worklytics — APPROVED** [review]

File: java/core/src/test/java/co/worklytics/psoxy/rules/RulesBaseTestCase.java [hunk 1]
```diff
@@ -153,4 +154,35 @@ protected void assertContains(String content, String... shouldContain) {
...
+    @SneakyThrows
+    protected void assertUrlWithQueryParamsBlocked(String url) {
+        assertFalse(sanitizer.isAllowed(new URL(url + "?param=value")), "query param allowed");
+        assertFalse(sanitizer.isAllowed(new URL(url + "?param=value&param2=value2")), "multiple query params allowed");
+    }
+
+    @SneakyThrows
+    protected void assertUrlWithSubResourcesAllowed(String url) {
+        assertTrue(sanitizer.isAllowed(new URL(url + "/dsfasdf")), "single param blocked");
+        assertTrue(sanitizer.isAllowed(new URL(url + "/asdfasdf/asdfasdf")), "multiple params blocked");
```
Comment (jlorper) [inline java/core/src/test/java/co/worklytics/psoxy/rules/RulesBaseTestCase.java:174]: Improve readability and nicer for OSS project. Same below.

```suggestion
        assertTrue(sanitizer.isAllowed(new URL(url + "/anypath")), "path blocked");
        assertTrue(sanitizer.isAllowed(new URL(url + "/anypath/anysubpath")), "path with subpath blocked");
```

File: java/core/src/test/java/co/worklytics/psoxy/rules/RulesBaseTestCase.java [hunk 2]
```diff
@@ -153,4 +154,35 @@ protected void assertContains(String content, String... shouldContain) {
...
+
+    @SneakyThrows
+    protected void assertUrlWithSubResourcesBlocked(String url) {
+        assertTrue(sanitizer.isAllowed(new URL(url + "/dsfasdf")), "single param blocked");
+        assertTrue(sanitizer.isAllowed(new URL(url + "/asdfasdf/asdfasdf")), "multiple params blocked");
+    }
+
+    @SneakyThrows
+    protected void assertUrlBlocked(String url) {
+        assertFalse(sanitizer.isAllowed(new URL(url)), "rules allowed url that should be blcoked: " + url);
```
Comment (jlorper) [inline java/core/src/test/java/co/worklytics/psoxy/rules/RulesBaseTestCase.java:185]:
```suggestion
        assertFalse(sanitizer.isAllowed(new URL(url)), "rules allowed url that should be blocked: " + url);
```

File: java/core/src/test/java/co/worklytics/psoxy/rules/google/DirectoryTests.java [hunk 3]
```diff
@@ -90,21 +90,26 @@ void groups() {
     @SneakyThrows
     @Test
     void group() {
+        String groupEndpoint = "https://admin.googleapis.com/admin/directory/v1/groups/asdfas";
```
Comment (jlorper) [inline java/core/src/test/java/co/worklytics/psoxy/rules/google/DirectoryTests.java:93]:
```suggestion
        String groupEndpoint = "https://admin.googleapis.com/admin/directory/v1/groups/an-arbitrary-subresource";
```

File: java/core/src/test/java/co/worklytics/psoxy/rules/google/GDriveTests.java [hunk 4]
```diff
@@ -41,14 +42,17 @@ void files() {
...
 
         assertPseudonymized(sanitized, PII);
+
+        assertUrlWithQueryParamsAllowed(endpoint);
     }
 
     @SneakyThrows
     @Test
     void revisions() {
+        String endpoint = "http://www.googleapis.com/drive/v2/files/Asdfasdfas/revisions";
```
Comment (jlorper) [inline java/core/src/test/java/co/worklytics/psoxy/rules/google/GDriveTests.java:55]: replace "Asdfasdfas" with something more readable

**Review by jlorper — CHANGES_REQUESTED** [review]
I'd do tests a little bit more readable so "garbage" used is more explicit

File: java/core/src/test/java/co/worklytics/psoxy/rules/google/DirectoryTests.java [hunk 5]
```diff
@@ -90,21 +90,26 @@ void groups() {
...
+
+        assertUrlWithSubResourcesAllowed(groupEndpoint);
+        assertUrlWithQueryParamsAllowed(groupEndpoint);
     }
 
     @SneakyThrows
     @Test
     void groupMembers() {
 
+        String membersEndpoint = "https://admin.googleapis.com/admin/directory/v1/groups/asdfas/members";
```
Comment (jlorper) [inline java/core/src/test/java/co/worklytics/psoxy/rules/google/DirectoryTests.java:112]:
```suggestion
        String membersEndpoint = "https://admin.googleapis.com/admin/directory/v1/groups/any-group-id/members";
```

File: java/core/src/test/java/co/worklytics/psoxy/rules/RulesBaseTestCase.java [hunk 6]
```diff
@@ -153,4 +154,35 @@ protected void assertContains(String content, String... shouldContain) {
...
+    @SneakyThrows
+    protected void assertUrlWithSubResourcesAllowed(String url) {
+        assertTrue(sanitizer.isAllowed(new URL(url + "/anypath")), "path blocked");
+        assertTrue(sanitizer.isAllowed(new URL(url + "/anypath/anysubpath")), "path with subpath blocked");
+    }
+
+    @SneakyThrows
+    protected void assertUrlWithSubResourcesBlocked(String url) {
+        assertTrue(sanitizer.isAllowed(new URL(url + "/anypath")), "subpath allowed");
+        assertTrue(sanitizer.isAllowed(new URL(url + "/anypath/anysubpath")), "2 subpathes allowed");
```
Comment (jlorper) [inline java/core/src/test/java/co/worklytics/psoxy/rules/RulesBaseTestCase.java:180]:
```suggestion
        assertTrue(sanitizer.isAllowed(new URL(url + "/anypath/anysubpath")), "2 subpaths allowed");
```

**Review by jlorper — APPROVED** [review]

**Pull request merged**